src/SDL_gpu_textengine.c

```diff
@@ -928,20 +929,40 @@ void TTF_DestroyGPUTextEngine(TTF_TextEngine *engine)
 
 TTF_TextEngine *TTF_CreateGPUTextEngine(SDL_GPUDevice *device)
 {
+    SDL_PropertiesID props = SDL_CreateProperties();
+    if (props == 0) {
+        SDL_SetError("Failed to create GPU text engine.");
+        return NULL;
+    }
+    SDL_SetPointerProperty(props, TTF_PROP_GPU_TEXT_ENGINE_DEVICE, device);
+
+    return TTF_CreateGPUTextEngineWithProperties(props);
+}
+
+TTF_TextEngine *TTF_CreateGPUTextEngineWithProperties(SDL_PropertiesID props)
+{
+    SDL_GPUDevice *device = SDL_GetPointerProperty(props, TTF_PROP_GPU_TEXT_ENGINE_DEVICE, NULL);
     if (!device) {
-        SDL_InvalidParamError("device");
+        SDL_SetError("Failed to create GPU text engine: Invalid device.");
         return NULL;
     }
 
     TTF_TextEngine *engine = (TTF_TextEngine *)SDL_malloc(sizeof(*engine));
     if (!engine) {
+        SDL_SetError("Failed to create GPU text engine.");
+        return NULL;
+    }
+
+    int atlas_texture_size = (int)SDL_GetNumberProperty(props, TTF_PROP_GPU_TEXT_ENGINE_ATLAS_TEXTURE_SIZE, 512);
+    if (atlas_texture_size <= 0) {
```

Comment:
Should zero mean the default?

Reply:
Is there any reason people will want to set the property to 0 particularly? Because otherwise I think handling 0 as an error might help to track potential bugs in code.